include/painless/parameter.h

```diff
@@ -31,38 +88,66 @@ inline std::string get_base_path() {
 namespace parser {
 
 template <typename T>
-inline bool from_string(const std::string& input, T& value) {
+inline bool from_string(const std::string&, T&, std::false_type) {
+  static_assert(detail::has_istream_operator<T&>::value,
+                "Type does not support the >> istream operator");
+  return false;
+}
+
+template <typename T>
+inline bool from_string(const std::string& input, T& value, std::true_type) {
   std::istringstream ss(input);
   return static_cast<bool>(ss >> value);
 }
 
-inline bool from_string(const std::string& input, bool& value) {
+inline bool from_string(const std::string& input, bool& value, std::true_type) {
   std::istringstream ss(input);
   return static_cast<bool>(ss >> std::boolalpha >> value);
 }
 
-inline bool from_string(const std::string& input, std::string& value) {
+inline bool from_string(const std::string& input,
+                        std::string& value,
+                        std::true_type) {
   value = input;
   return true;
 }
 
+template <typename T>
+inline bool from_string(const std::string& input, T& value) {
+  return from_string(input, value,
+                     typename detail::has_istream_operator<T&>::type{});
+}
+
 }  // namespace parser
 
 namespace printer {
 
 template <typename T>
-inline std::string to_string(const T& value) {
+inline std::string to_string(const T&, std::false_type) {
+  static_assert(detail::has_istream_operator<const T&>::value,
+                "Type does not support the << ostream operator");
+  return std::string{};
+}
+
+template <typename T>
+inline std::string to_string(const T& value, std::true_type) {
   std::stringstream ss;
   ss << value;
   return ss.str();
 }
 
-inline std::string to_string(const bool& value) {
+inline std::string to_string(const bool& value, std::true_type) {
   std::stringstream ss;
   ss << std::boolalpha << value;
   return ss.str();
 }
 
+template <typename T>
+inline std::string to_string(const T& value) {
+  return to_string(value,
+                   typename detail::has_ostream_operator<const T&>::type{});
```

Comment:
Can you remind me why tag dispatch is required/preferred here? Couldn't we simply add a
```c++
static_assert(detail::has_istream_operator<const T&>::value,
              "Type does not support the << ostream operator"); 
```
in the class scope of `Parameter<T>`?

Reply:
That would also be possible.  I thought that having check at the point of use is maybe more reusable.